src/mixins/PreviewUrl.js

```diff
@@ -63,12 +63,17 @@ export default {
 		 * @returns {String} the absolute url
 		 */
 		getPreviewIfAny({ fileid, filename, hasPreview, davPath }) {
+			const searchParams = `fileId=${fileid}`
+				+ `&x=${Math.floor(screen.width * devicePixelRatio)}`
+				+ `&y=${Math.floor(screen.height * devicePixelRatio)}`
+				+ '&a=true'
+
 			if (hasPreview) {
 				// TODO: find a nicer standard way of doing this?
 				if (isPublic()) {
-					return generateUrl(`/apps/files_sharing/publicpreview/${getToken()}?fileId=${fileid}&file=${encodeFilePath(filename)}&x=${screen.width}&y=${screen.height}&a=true`)
+					return generateUrl(`/apps/files_sharing/publicpreview/${getToken()}?file=${encodeFilePath(filename)}&${searchParams}`)
```

Comment:
Could even just be set as params with `searchParams` being an object :)

```suggestion
					return generateUrl(`/apps/files_sharing/publicpreview/${getToken()}`, {
						file: encodeFilePath(filename),
						...searchParams,
					})
```

Reply:
Reminder to self: `generateUrl` does some encoding

Reply:
right, feel free to ignore then :)